website/docs/guides/getting-started.html.markdown

```diff
@@ -54,450 +49,603 @@ orchestration with Terraform presents a few benefits.
+}
+```
 
-* provider.kubernetes: version = "~> 1.10.1"
+For short-lived authentication tokens, like those found in EKS, which [expire in 15 minutes](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#controlling-access-to-eks-clusters), an exec-based credential plugin can be used to ensure the token is always up to date:
 
-Terraform has been successfully initialized!
+```hcl
+data "aws_eks_cluster" "example" {
+  name = "example"
+}
 
-You may now begin working with Terraform. Try running "terraform plan" to see
-any changes that are required for your infrastructure. All Terraform commands
-should now work.
+data "aws_eks_cluster_auth" "example" {
+  name = "example"
+}
 
-If you ever set or change modules or backend configuration for Terraform,
-rerun this command to reinitialize your working directory. If you forget, other
-commands will detect it and remind you to do so if necessary.
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.example.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.example.certificate_authority[0].data)
+￼ exec {
+￼   api_version = "client.authentication.k8s.io/v1alpha1"
+￼   args        = ["eks", "get-token", "--cluster-name", var.cluster_name]
+￼   command     = "aws"
+￼ }
+}
 ```
 
-## Scheduling a Simple Application
+## Creating your first Kubernetes resources
 
-The main object in any Kubernetes application is [a Pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/#what-is-a-pod).
-Pod consists of one or more containers that are placed
-on cluster nodes based on CPU or memory availability.
+Once the provider is configured, you can apply the Kubernetes resources defined in you Terraform config file. The following is an example Terraform config file containing a few Kubernetes resources. We'll use [minikube](https://minikube.sigs.k8s.io/docs/start/) for the Kubernetes cluster in this example, but any Kubernetes cluster can be used. Ensure that a Kubernetes cluster of some kind is running before applying the example config below.
 
-Here we create a pod with a single container running the nginx web server,
-exposing port 80 (HTTP) which can be then exposed
-through the load balancer to the real user.
-
-Unlike in this simple example you'd commonly run more than
-a single instance of your application in production to reach
-high availability and adding labels will allow Kubernetes to find all
-pods (instances) for the purpose of forwarding the traffic
-to the exposed port.
+This configuration will create a scalable Nginx Deployment with 2 replicas. It will expose the Nginx frontend using a Service of type NodePort, which will make Nginx accessible via the public IP of the node running the containers.
 
 ```hcl
-resource "kubernetes_pod" "nginx" {
-  metadata {
-    name = "nginx-example"
-    labels = {
-      App = "nginx"
+terraform {
+  required_providers {
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "9.9.9"
```

Comment:
I think we want a `>= 2.0.0` constraint here rather than `9.9.9`.

Reply:
oops, good catch! That was from local testing.